src/WordPress/Hooks.php

```diff
@@ -82,47 +79,17 @@ public function initProxy()
 
     public function activate()
     {
-        $this->checkVersionCompatibility();
-        $this->checkDependenciesExist();
```

Comment:
I think its fine.  `activate()` documents the function gets called during activation and the `wp_die()` errors tell you its version related.  I was trying to avoid functions in Hooks.php that aren't directly mapped to an actual WordPress hook.